test/integration/controllermanager/quota/quota_test.go

```diff
@@ -0,0 +1,162 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package quota_test
+
+import (
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gutil "github.com/gardener/gardener/pkg/utils"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Quota controller tests", func() {
+	var (
+		providerType string
+		resourceName string
+		objectKey    client.ObjectKey
+
+		secret        *corev1.Secret
+		quota         *gardencorev1beta1.Quota
+		secretBinding *gardencorev1beta1.SecretBinding
+	)
+
+	BeforeEach(func() {
+		providerType = "provider-type"
+		resourceName = "test-" + gutil.ComputeSHA256Hex([]byte(CurrentSpecReport().LeafNodeLocation.String()))[:8]
+		objectKey = client.ObjectKey{Namespace: testNamespace.Name, Name: resourceName}
+
+		secret = &corev1.Secret{
+			ObjectMeta: kutil.ObjectMetaFromKey(objectKey),
+		}
+
+		quota = &gardencorev1beta1.Quota{
+			ObjectMeta: kutil.ObjectMetaFromKey(objectKey),
+			Spec: gardencorev1beta1.QuotaSpec{
+				Scope: corev1.ObjectReference{
+					APIVersion: "v1",
+					Kind:       "Secret",
+				},
+			},
+		}
+
+		secretBinding = &gardencorev1beta1.SecretBinding{
+			ObjectMeta: kutil.ObjectMetaFromKey(objectKey),
+			Provider: &gardencorev1beta1.SecretBindingProvider{
+				Type: providerType,
+			},
+			SecretRef: corev1.SecretReference{
+				Name:      resourceName,
+				Namespace: testNamespace.Name,
+			},
+			Quotas: []corev1.ObjectReference{
+				{
+					Name:      resourceName,
+					Namespace: testNamespace.Name,
+				},
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		By("Create Secret")
+		Expect(testClient.Create(ctx, secret)).To(Succeed())
+		log.Info("Created Secret for test", "secret", client.ObjectKeyFromObject(secret))
+
+		By("Create Quota")
+		Expect(testClient.Create(ctx, quota)).To(Succeed())
+		log.Info("Created Quota for test", "quota", client.ObjectKeyFromObject(quota))
+
+		DeferCleanup(func() {
+			By("Delete Quota")
+			Expect(testClient.Delete(ctx, quota)).To(Or(Succeed(), BeNotFoundError()))
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(quota), quota)
+			}).Should(BeNotFoundError())
+		})
+
+		DeferCleanup(func() {
+			By("Delete Secret")
+			Expect(testClient.Delete(ctx, secret)).To(Or(Succeed(), BeNotFoundError()))
+		})
```

Comment:
Nit: secret is not cleaned up if quota creation fails, you could move this block up after creating the secret